.circleci/osx_install_dependencies.sh

```diff
@@ -50,33 +50,65 @@ if [ ! -f /usr/local/lib/libz3.a ] # if this file does not exists (cache was not
 then
   brew update
   brew upgrade
-  brew install boost
   brew install cmake
   brew install wget
   brew install coreutils
   brew install diffutils
   brew install grep
-  ./scripts/install_obsolete_jsoncpp_1_7_4.sh
+
+  # writing to /usr/local/lib need administrative privileges.
+  sudo ./scripts/install_obsolete_jsoncpp_1_7_4.sh
+
+  # boost
+  boost_version="1.84.0"
+  boost_package="boost_${boost_version//./_}.tar.bz2"
+  boost_dir="boost_${boost_version//./_}"
+  wget "https://boostorg.jfrog.io/artifactory/main/release/$boost_version/source/$boost_package"
+  tar xf "$boost_package"
+  rm "$boost_package"
+  cd "$boost_dir"
+  ./bootstrap.sh --with-toolset=clang --with-libraries=thread,system,filesystem,program_options,serialization,test
+  sudo ./b2 -a address-model=64 architecture=arm+x86 install
+  cd ..
+  sudo rm -rf "$boost_dir"
 
   # z3
   z3_version="4.12.1"
-  z3_dir="z3-${z3_version}-x64-osx-10.16"
-  z3_package="${z3_dir}.zip"
-  wget "https://github.com/Z3Prover/z3/releases/download/z3-${z3_version}/${z3_package}"
-  validate_checksum "$z3_package" 7601f844de6d906235140d0f76cca58be7ac716f3e2c29c35845aa24b24f73b9
-  unzip "$z3_package"
+  z3_dir="z3-z3-$z3_version"
+  z3_package="z3-$z3_version.tar.gz"
+  wget "https://github.com/Z3Prover/z3/archive/refs/tags/$z3_package"
+  validate_checksum "$z3_package" a3735fabf00e1341adcc70394993c05fd3e2ae167a3e9bb46045e33084eb64a3
+  tar xf "$z3_package"
   rm "$z3_package"
-  cp "${z3_dir}/bin/libz3.a" /usr/local/lib
-  cp "${z3_dir}/bin/z3" /usr/local/bin
-  cp "${z3_dir}/include/"* /usr/local/include
-  rm -r "$z3_dir"
+  cd "$z3_dir"
+  mkdir build
+  cd build
+  cmake -DCMAKE_OSX_ARCHITECTURES:STRING="x86_64;arm64" -DZ3_BUILD_LIBZ3_SHARED=false ..
+  make -j
+  sudo make install
+  cd ../..
+  rm -rf "$z3_dir"
 
   # evmone
   evmone_version="0.11.0"
-  evmone_package="evmone-${evmone_version}-darwin-x86_64.tar.gz"
-  wget "https://github.com/ethereum/evmone/releases/download/v${evmone_version}/${evmone_package}"
-  validate_checksum "$evmone_package" 83ed20676681d9a31bd30cac399ab7c615ccab8adb8087cc2c7e9cd22b4d2efc
-  tar xzpf "$evmone_package" -C /usr/local
-  rm "$evmone_package"
-
+  if [[ $(uname -m) == 'arm64' ]]
+  then
+    # evmone does not provide any builds for apple silicon yet. so lets just build it locally.
+    git clone https://github.com/ethereum/evmone.git
+    cd evmone
+    git checkout "v${evmone_version}"
+    git submodule update --init
+    cmake -S . -B build
+    cmake --build build --parallel
```

Comment:
We have `MAKEFLAGS` set in CI config for all jobs according to the number of cores so `--parallel` should not be necessary. I'm actually not sure how they interact so I'd be worried about `--parallel` overwriting it with a different value that may or may not be autodetected properly.

Reply:
I just removed `--parallel` here